build.rs

```diff
@@ -242,7 +240,28 @@ fn main() -> io::Result<()> {
         .must_use_type("AKRESULT")
         .enable_cxx_namespaces()
         .parse_callbacks(Box::new(bindgen::CargoCallbacks))
-        .layout_tests(false)
+        .layout_tests(false);
+
+    #[cfg(any(target_os = "windows", target_os = "linux"))]
+    let bindings_builder = bindings_builder
+        .clang_arg("-x")
+        .clang_arg("c++")
+        .clang_arg("-std=c++14");
+
+    #[cfg(target_os = "macos")]
+    let bindings_builder = bindings_builder
+        .clang_args(&["-x", "c++"])
+        .clang_args(&["-std=c++17"])
+        .clang_args(&[
+            "-isysroot",
+            "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks",
+        ])
+        .clang_args(&[
+            "-isysroot",
+            "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk",
+        ]);
```

Comment:
Instead of adding target-specific branching here, could you maybe create a `bindings_cc_platform_specifics()` function in windows.rs, unix.rs and macos.rs that encapsulate this (mimicking what I do for the compilation of the stream manager)? Windows and Unix should have the same implementation, but macos would have the c++17 and `isysroot` arguments.  BTW, `-x` and `c++` can be part of the common trunk above ;)   

Feel free to ask if somethings is not clear!

Reply:
Actually, `std=c++14` should be placed in the common trunk, windows.rs and unix.rs should do nothing in this function, and macos.rs should only add the `isysroot` arguments